ExRam.Gremlinq.Core.Tests/GroovySerializationTest.cs

```diff
@@ -252,6 +295,288 @@ public void AddV_without_model()
+                    .IgnoreOnUpdate(p => p.Age)
+                    .IgnoreAlways(p => p.Name)
+                )
+                .ConfigureElement<WorksFor>(builder => builder
+                    .IgnoreAlways(p => p.From)
+                    .IgnoreOnUpdate(p => p.Role)
+                )
+            ));
+
+            g
+                .V<Person>()
+                .UpdateV(person)
+                .OutE<WorksFor>()
+                .UpdateE(worksFor)
+                .Should()
+                .SerializeToGroovy<TVisitor>("g.V().hasLabel(_a).sideEffect(__.properties(_b, _c).drop()).property(single, _b, _d).property(single, _c, _e).outE(_f).sideEffect(__.properties(_g).drop()).property(_g, _h)")
+                .WithParameters(nameof(Person),
+                    nameof(Person.Gender), nameof(Person.RegistrationDate),
+                    person.Gender, person.RegistrationDate,
+                    nameof(WorksFor), nameof(WorksFor.To),
+                    worksFor.To);
+        }
+
+        [Fact]
+        public void Update_Vertex_And_Edge_No_Config()
+        {
+            var now = DateTimeOffset.UtcNow;
+            var edgeNow = DateTime.UtcNow;
+            var person = new Person { Age = 21, Gender = Gender.Male, Name = "Marko", RegistrationDate = now };
+            var worksFor = new WorksFor { From = edgeNow, To = edgeNow, Role = "Admin" };
+
+            g
+                .V<Person>()
+                .UpdateV(person)
+                .OutE<WorksFor>()
+                .UpdateE(worksFor)
+                .Should()
+                .SerializeToGroovy<TVisitor>("g.V().hasLabel(_a).sideEffect(__.properties(_b, _c, _d, _e).drop()).property(single, _b, _f).property(single, _c, _g).property(single, _d, _h).property(single, _e, _i).outE(_j).sideEffect(__.properties(_k, _l, _m).drop()).property(_k, _n).property(_l, _o).property(_m, _o)")
+                .WithParameters(nameof(Person),
+                    nameof(Person.Age), nameof(Person.Name), nameof(Person.Gender), nameof(Person.RegistrationDate),
+                    person.Age, "Marko", person.Gender, person.RegistrationDate,
+                    nameof(WorksFor), nameof(WorksFor.Role), nameof(WorksFor.From), nameof(WorksFor.To),
+                    worksFor.Role, worksFor.From);
+        }
+
+        [Fact]
+        public void ReplaceV()
+        {
+            var now = DateTimeOffset.UtcNow;
+            var id = Guid.NewGuid();
+            var person = new Person { Id = id, Age = 21, Gender = Gender.Male, Name = "Marko", RegistrationDate = now };
+            g
+                .ReplaceV(person)
+                .Should()
+                .SerializeToGroovy<TVisitor>("g.V(_a).sideEffect(__.properties(_b, _c, _d, _e, _f).drop()).property(id, _a).property(single, _b, _g).property(single, _c, _h).property(single, _d, _i).property(single, _e, _j)")
+                .WithParameters(id, nameof(Person.Age), nameof(Person.Name), nameof(Person.Gender),
```

Comment:
Sorry for delay in review. This looks good to me, I just don't get the difference between Update and Replace, especially since this won't work, at least not in CosmosDB and the generic Tinkerpop Server. You just can't overwrite ids. So, what's the actual point of Replace?

Reply:
Good point on the Id property. Originally, I was thinking that the user would just set up the property as read only, but in reality, we should never attempt to replace it. So, I updated the code to default "Id" to IgnoreOnUpdate.

Also, since we're using PropertyInfos as a key to the dictionary, I have to make sure that the property infos match exactly with the key, so I had to update the binding flags in ObjectExtensions.Serialize to match up with what you're doing in GraphElementPropertiesModel.FromGraphElementModels.

Reply:
Alright. I think this is good to be merged. The only thing that's unclear to me is what should happen on UpdateV/E when the passed entity has its Id set. It'll be ignored, but there's some different intention of the user to be assumed...should it throw instead?

Reply:
I know in my usage pattern, I would normally be passing in a fully populated object for update, including the Id. In my case, I'd prefer it to just ignore the id and not throw because I wouldn't want to do an extra step of creating a new object just for the update operation without the id populated.

Reply:
Alright, let's go with this behaviour for the time being. I rebased and force-pushed the branch back to your repo. I'm mergin now but if needed, you might wanna do a fetch.